### PR TITLE
Split afs in modules

### PR DESCRIPTION
**Pull request by yichen88**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#934 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [X] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: scripting/pom.xml [hunk 1]
```diff
@@ -11,127 +11,21 @@
...
-            <groupId>${project.groupId}</groupId>
-            <artifactId>powsybl-tools</artifactId>
-            <version>${project.version}</version>
-            <scope>test</scope>
-            <type>test-jar</type>
-        </dependency>
-    </dependencies>
+    <modules>
+        <module>script-afs</module>
+        <module>script-extension</module>
```
Comment (mathbagu) [inline scripting/pom.xml]: Should we keep **scripting** in the name if these modules?

File: scripting/script-afs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        <version>3.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>powsybl-script-afs</artifactId>
+    <name>Scripting AFS</name>
+    <description>A tool to run scripts over IIDM and AFS</description>
+
+    <properties>
+        <jline2.version>2.12</jline2.version>
```
Comment (mathbagu) [inline scripting/script-afs/pom.xml]: Versions should be moved to the main pom.xml

File: scripting/script-afs/pom.xml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <scope>provided</scope>
+        </dependency>
```
Comment (mathbagu) [inline scripting/script-afs/pom.xml]: Do we need to declared this dependency? Should be moved in a dedicated section (before the test section)

File: scripting/script-extension/pom.xml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        <version>3.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>powsybl-script-extension</artifactId>
+    <name>Script extension</name>
+    <description>Groovy script extension</description>
+
+    <properties>
+        <jline2.version>2.12</jline2.version>
```
Comment (mathbagu) [inline scripting/script-extension/pom.xml]: Same remark here

File: scripting/script-extension/pom.xml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <scope>provided</scope>
+        </dependency>
```
Comment (mathbagu) [inline scripting/script-extension/pom.xml]: Same remark here

File: scripting/script-extension/pom.xml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    <dependencies>
+        <!-- Compilation dependencies -->
+        <dependency>
+            <groupId>org.codehaus.groovy</groupId>
+            <artifactId>groovy-all</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jline</groupId>
+            <artifactId>jline</artifactId>
+            <version>${jline2.version}</version>
```
Comment (mathbagu) [inline scripting/script-extension/pom.xml]: You should declare this dependency in the dependency management section of the pom.xml

File: scripting/pom.xml (hunk 1 above)
Comment (geofjamg) [reply scripting/pom.xml]: Done

**Comment by MioRtia** [comment]
Since it is a **Breaking Change**, could you please update the migration guide?

**Comment by MioRtia** [comment]
Change the version of your new modules in `3.1.0-SNAPSHOT` to prevent failing

**Pull request merged**